src/instance/methods/features.js

```diff
@@ -0,0 +1,34 @@
+// Import formats.
+import GeoJSON from 'ol/format/GeoJSON';
+import WKT from 'ol/format/WKT';
+
+// Import the default projection configuration
+import projection from '../../projection';
+
+// Read features from Well Known Text (WKT).
+function readFeaturesWKT(wkt) {
+  const isMultipart = wkt.includes('MULTIPOINT')
+    || wkt.includes('MULTILINESTRING')
+    || wkt.includes('MULTIPOLYGON')
+    || wkt.includes('GEOMETRYCOLLECTION');
+  return isMultipart
+    ? new WKT({ splitCollection: true }).readFeatures(wkt, projection)
+    : [new WKT().readFeature(wkt, projection)];
+}
+
+// Read features from GeoJSON.
+function readFeaturesGeoJSON(geojson) {
+  return (new GeoJSON()).readFeatures(geojson, projection);
+}
+
+// Read features from various formats.
+export default function readFeatures(type, input) {
+  let features;
+  if (type.toLowerCase() === 'geojson') {
+    features = readFeaturesGeoJSON(input);
+  }
+  if (type.toLowerCase() === 'wkt') {
+    features = readFeaturesWKT(input);
+  }
+  return features;
```

Comment:
Nitpick: avoid unnecessary variable/assignments and make fallback return value explicit;

```suggestion
  if (type.toLowerCase() === 'geojson') {
    return readFeaturesGeoJSON(input);
  }
  if (type.toLowerCase() === 'wkt') {
    return readFeaturesWKT(input);
  }
  return undefined;
```

Reply:
That's better I agree! Fixup: https://github.com/farmOS/farmOS-map/pull/194/commits/c30bb5a2afeafc3c8cfccc41ab5e473fe54dec43